File: ProjetSopraFinal/src/main/java/com/example/demo/model/Adresse.java
package com.example.demo.model;

import javax.persistence.Embeddable;

import com.example.demo.jsonview.JsonViews;
import com.fasterxml.jackson.annotation.JsonView;

@Embeddable
public class Adresse {
	
	private Integer numero;
	private String rue;
	private Integer cp;
	private String ville;
	private String pays;
	
	
	public Adresse() {

	}


	@JsonView(JsonViews.Common.class)
	public Integer getNumero() {
		return numero;
	}


	public void setNumero(Integer numero) {
		this.numero = numero;
	}


	@JsonView(JsonViews.Common.class)
	public String getRue() {
		return rue;
	}


	public void setRue(String rue) {
		this.rue = rue;
	}


	@JsonView(JsonViews.Common.class)
	public Integer getCp() {
		return cp;
	}


	public void setCp(Integer cp) {
		this.cp = cp;
	}


	@JsonView(JsonViews.Common.class)
	public String getVille() {
		return ville;
	}


	public void setVille(String ville) {
		this.ville = ville;
	}


	@JsonView(JsonViews.Common.class)
	public String getPays() {
		return pays;
	}


	public void setPays(String pays) {
		this.pays = pays;
	}
	
	

}

File: AngularSopraFinal/src/app/model/adresse.ts
export class Adresse {

  constructor(private _numero?, private _rue?, private _cp?, private _ville?, private _pays?) {
  }

  public get numero(): number {
    return this._numero;
  }
  public set numero(_numero: number) {
    this._numero = _numero;
  }

  public get rue(): string {
    return this._rue;
  }
  public set rue(_rue: string) {
    this._rue = _rue;
  }

  public get cp(): number {
    return this._cp;
  }
  public set cp(_cp: number) {
    this._cp = _cp;
  }

  public get ville(): string {
    return this._ville;
  }
  public set ville(_ville: string) {
    this._ville = _ville;
  }

  public get pays(): string {
    return this._pays;
  }
  public set pays(_pays: string) {
    this._pays = _pays;
  }

}
